02-hello-accessibility/projects/starter/app/src/main/java/com/raywenderlich/android/tacotuesday/detail/RecipeDetailFragment.kt

```diff
@@ -0,0 +1,169 @@
+    recipeDetailBaseLayer.movementMethod = LinkMovementMethod.getInstance()
+    recipeDetailCondiment.text = recipe.condiment.toMarkdownOrDefaultText(
+        getString(R.string.recipe_detail_condiment))
+    recipeDetailCondiment.movementMethod = LinkMovementMethod.getInstance()
+    recipeDetailMixin.text = recipe.mixin.toMarkdownOrDefaultText(
+        getString(R.string.recipe_detail_mixin))
+    recipeDetailMixin.movementMethod = LinkMovementMethod.getInstance()
+    recipeDetailSeasoning.text = recipe.seasoning.toMarkdownOrDefaultText(
+        getString(R.string.recipe_detail_seasoning))
+    recipeDetailSeasoning.movementMethod = LinkMovementMethod.getInstance()
+    recipeDetailShell.text = recipe.shell.toMarkdownOrDefaultText(
+        getString(R.string.recipe_detail_shell))
+    recipeDetailShell.movementMethod = LinkMovementMethod.getInstance()
+    recipeDetailCreditText.movementMethod = LinkMovementMethod.getInstance()
+    recipeDetailComplimentTheChef.setOnClickListener {
+      binding.recipeDetailErrorView.visibility = View.VISIBLE
+    }
+  }
+
+  private fun String?.toMarkdownOrDefaultText(label: String) = if (this != null) {
+    markwon.toMarkdown(this)
+  } else {
+    getString(R.string.recipe_detail_default_text, label)
+  }
+
+  private fun showEditableFields(recipe: Recipe) = with(binding) {
+    recipeDetailTryDiscardButton.setImageDrawable(ResourcesCompat.getDrawable(resources,
+        R.drawable.ic_baseline_thumb_down_24, requireContext().theme))
+    recipeDetailTryDiscardButton.setOnClickListener {
+      viewModel.discardRecipe(recipe, viewModel.viewModelScope)
+    }
+    recipeNachoRating.rating = recipe.nachoRating ?: 0
+    recipeSpiceRating.rating = recipe.spiceRating ?: 0
+    recipeDetailNotesEditText.setText(recipe.notes)
+    recipeDetailMadeIt.isChecked = recipe.hasMade
+    recipeDetailMadeIt.setOnCheckedChangeListener { _, _ -> activity?.onBackPressed() }
+    recipeNachoRatingContainer.visibility = View.VISIBLE
+    recipeSpiceRatingContainer.visibility = View.VISIBLE
+    recipeDetailNotesEditText.visibility = View.VISIBLE
+    recipeDetailNotesLabel.visibility = View.VISIBLE
+    recipeDetailMadeIt.visibility = View.VISIBLE
+    recipeDetailMadeItLabel.visibility = View.VISIBLE
+  }
+
+  private fun hideEditableFields(recipe: Recipe?) = with(binding) {
+    recipeDetailTryDiscardButton.setImageDrawable(ResourcesCompat.getDrawable(resources,
+        R.drawable.ic_baseline_thumb_up_24, requireContext().theme))
+    recipeDetailTryDiscardButton.setOnClickListener {
+      viewModel.tryRecipe(recipe, viewModel.viewModelScope)
+    }
+    recipeNachoRatingContainer.visibility = View.GONE
+    recipeSpiceRatingContainer.visibility = View.GONE
+    recipeDetailNotesEditText.visibility = View.GONE
+    recipeDetailNotesLabel.visibility = View.GONE
+    recipeDetailMadeIt.visibility = View.GONE
+    recipeDetailMadeItLabel.visibility = View.GONE
+  }
+
+  private fun getMarkwon(): Lazy<Markwon> {
```

Comment:
I feel like this would make more sense to have closer to where the `markwon by lazy` bit is set up, unless there's something I'm missing in terms of why it's down here.

Reply:
Personal preference of putting private things in the bottom. I have no problem moving it up